### PR TITLE
Get rid of extra Soci thread (RIPD-862).

### PR DESCRIPTION
**Pull request by rec**
Reviewers: @seelabs @ximinez

### CODE REVIEW

**Comment by seelabs** [comment]
@rec: This looks good. Nice job! I'm not able to recreate the Travis build error, I believe it's spurious. Assuming that's not an issue :+1: from me.

File: src/ripple/app/data/SociDB.cpp [hunk 1]
```diff
@@ -172,112 +172,78 @@ namespace {
...
-    int sqliteWALHook (void* s, sqlite_api::sqlite3*,
-                       const char* dbName, int walSize);
+    ~WALCheckpointer () override = default;
 
 private:
-    void runCheckpoint (const char* db, int walSize);
-    void run ();
-    void checkpoint ();
-
     using LockType = std::mutex;
```
Comment (seelabs) [inline src/ripple/app/data/SociDB.cpp]: I misnamed this. It should be `MutexType`. Not part of this PR, but you could rename this while you're mucking with this file.

Comment (HowardHinnant) [inline src/ripple/app/data/SociDB.cpp]: :+1:

Comment (vinniefalco) [inline src/ripple/app/data/SociDB.cpp]: We should abolish the practice of these pointless nested types like `LockType` or `MutexType`

Comment (rec) [inline src/ripple/app/data/SociDB.cpp]: killed.

**Comment by ximinez** [comment]
Looks good, and builds fine for me. :+1:

**Comment by vinniefalco** [comment]
Suggested:
`Remove unused soci thread (RIPD-682)`
